src/main.rs

```diff
@@ -86,21 +88,25 @@ fn download_datasets(datasets: &[&str], base_url: &str) {
     }
 }
 
+#[allow(unused_must_use)]
 fn unzip_datasets(datasets: &[&str]) {
     for filename in datasets {
         let mut file_handle = File::open(&format!("assets/{}", filename))
             .unwrap();
         let mut in_file: Vec<u8> = Vec::new();
+        let mut decompressed_file: Vec<u8> = Vec::new();
         
-        file_handle.read_to_end(&mut in_file);
+        file_handle.read_to_end(&mut in_file).unwrap();
+
+        let mut decoder = GzDecoder::new(in_file.as_slice()).unwrap();
 
-        let unzipped_data = inflate_bytes_zlib(in_file.as_slice()).unwrap();
+        decoder.read_to_end(&mut decompressed_file).unwrap();
 
```

Comment:
There are a lots of `unwrap()`s, I think an `expect(...)` with some descriptive error text would be nicer.